runtime/lua/vim/_editor.lua

```diff
@@ -1107,13 +1107,26 @@ end
 
 function vim._init_default_autocmds()
   local nvim_terminal_augroup = vim.api.nvim_create_augroup('nvim_terminal', {})
-  vim.api.nvim_create_autocmd({ 'bufreadcmd' }, {
+  vim.api.nvim_create_autocmd({ 'BufReadCmd' }, {
     pattern = 'term://*',
     group = nvim_terminal_augroup,
     nested = true,
     command = "if !exists('b:term_title')|call termopen(matchstr(expand(\"<amatch>\"), '\\c\\mterm://\\%(.\\{-}//\\%(\\d\\+:\\)\\?\\)\\?\\zs.*'), {'cwd': expand(get(matchlist(expand(\"<amatch>\"), '\\c\\mterm://\\(.\\{-}\\)//'), 1, ''))})",
   })
-  vim.api.nvim_create_autocmd({ 'cmdwinenter' }, {
+  vim.api.nvim_create_autocmd({ 'TermClose' }, {
+    group = nvim_terminal_augroup,
+    desc = 'Automatically close terminal buffers when started with no arguments and exiting without an error',
+    callback = function(args)
+      if vim.v.event.status == 0 then
+        local info = vim.api.nvim_get_chan_info(vim.bo[args.buf].channel)
+        local argv = info.argv or {}
+        if #argv == 1 and argv[1] == vim.o.shell then
+          vim.cmd({ cmd = 'bdelete', args = { args.buf }, bang = true })
```

Comment:
This will close the current split window (if any). That's probably unwanted in many cases, so we need either of:

- open `:terminal` in split by default
- add a `:bdelete` variant that doesn't close the current split (or hack it here using `b#`) https://github.com/neovim/neovim/issues/2434

Reply:
I made https://github.com/neovim/neovim/pull/15442 a while ago for the first option. Even though it would probably break a lot of things, I think I’d still prefer the split-by-default behavior 😬

Reply:
Do you mean https://github.com/neovim/neovim/pull/15427 ? Unless it breaks fugitive I think we should try it.

Reply:
Oops, yes, thanks. Updated my original comment.